Automattic/pocket-casts-android
# Cache playing episode feature flag + advanced setting

## Description

This is a preparatory PR  that

1. Adds feature flag for caching the entire streaming episode to minimize streaming while playing (this is different from on-the-fly caching introduced earlier).
2. Adds advanced setting if the above feature flag is enabled (it is disabled by default, we can revisit the default state, and summary text later).
3. Extracts data source factory to helper singleton so that they can be reused later to cache entire episode.

## Testing Instructions
1. Install the app
2. Go to `Profile` -> `Settings`-> `Advanced`
3. ✅ Notice there is an advanced setting to enable caching the playing episode 
4. Toggle the setting
5. ✅ Notice that `settings_advanced_cache_entire_playing_episode` is tracked
6. Go to `Profile` -> `Settings`->  `Beta features` 
7. ✅ Disable "Cache entire playing episode" feature
8. Go to `Profile` -> `Settings`-> `Advanced`
9. ✅ Notice there is no advanced setting to enable caching the playing episode 

## Screenshots or Screencast 
<img width=200 src="https://github.com/Automattic/pocket-casts-android/assets/1405144/71d31b79-7046-41dd-bfdf-469125d67faf"/>


## Checklist
 
#### I have tested any UI changes...
